### ISSUE TITLE
assertionResults empty

### ISSUE THREAD

**Issue by nhomble** [marilari88/neotest-vitest#59]
I am trying to figure out why this happens, but my nil check on my [fork](https://github.com/nhomble/neotest-vitest) has unblocked me.

Unless other folks have seen this before, I am going to wait to submit a pull request once I understand the root cause a little better to justify the check.

**Comment by nhomble** [comment]
FWIW the types claim we shouldn't get `nil`, https://github.com/vitest-dev/vitest/blob/main/packages/vitest/src/node/reporters/json.ts#L47

I'll keep this open in case anyone else in the community has encountered this.

**Comment by nhomble** [comment]
🤦 when you're on an ancient version of `vitest` then yeah it makes sense certain fields were added later: https://github.com/vitest-dev/vitest/blob/v0.3.5/packages/vitest/src/node/reporters/json.ts#L9

Since the plugin works without this additional metadata, I think it's nice-to-have but realistically any actual user of `vitest` is going to be on version with this in the json reporter
